src/main/java/org/folio/helper/CheckinReceivePiecesHelper.java

```diff
@@ -308,52 +329,82 @@ private List<Piece> getSuccessfullyProcessedPieces(String poLineId, Map<String,
+   * @param poLine                       PO Line record representation from storage
    * @return completable future holding calculated PO Line's receipt status
    */
-  private Future<ReceiptStatus> calculatePoLineReceiptStatus(List<Piece> byPoLine, PoLine poLine, List<Piece> pieces) {
-    long expectedPiecesQuantity = byPoLine.stream()
+  private ReceiptStatus calculatePoLineReceiptStatus(List<Piece> piecesByPoLine, List<Piece> piecesSuccessfullyProcessed, PoLine poLine) {
+    logger.info("calculatePoLineReceiptStatus:: piecesByPoLine: {}", piecesByPoLine.size());
+    logger.info("calculatePoLineReceiptStatus:: pieces: {}", piecesSuccessfullyProcessed.size());
+    piecesByPoLine.forEach(v -> logger.info("byPoLine:: id: {}, status: {}", v.getId(), v.getReceivingStatus()));
+    piecesSuccessfullyProcessed.forEach(v -> logger.info("pieces:: id: {}, status: {}", v.getId(), v.getReceivingStatus()));
+    long expectedPiecesQuantity = piecesByPoLine.stream()
       .filter(piece -> EXPECTED_STATUSES.contains(piece.getReceivingStatus()))
       .count();
+    logger.info("calculatePoLineReceiptStatus:: expectedPiecesQuantity: {}", expectedPiecesQuantity);
     // Fully Received: If receiving and there is no expected piece remaining
+    logger.info("calculatePoLineReceiptStatus:: poLine.getCheckinItems(): {}", poLine.getCheckinItems());
+    logger.info("calculatePoLineReceiptStatus:: FULLY_RECEIVED: {}", !poLine.getCheckinItems().equals(Boolean.TRUE) && expectedPiecesQuantity == 0);
     if (!poLine.getCheckinItems().equals(Boolean.TRUE) && expectedPiecesQuantity == 0) {
-      return Future.succeededFuture(FULLY_RECEIVED);
+      return FULLY_RECEIVED;
     }
     // Partially Received: In case there is at least one successfully received piece
-    if (StreamEx.of(pieces).anyMatch(piece -> RECEIVED_STATUSES.contains(piece.getReceivingStatus()))) {
-      return Future.succeededFuture(PARTIALLY_RECEIVED);
+    logger.info("calculatePoLineReceiptStatus:: PARTIALLY_RECEIVED: {}", StreamEx.of(piecesSuccessfullyProcessed).anyMatch(piece -> RECEIVED_STATUSES.contains(piece.getReceivingStatus())));
+    if (StreamEx.of(piecesSuccessfullyProcessed).anyMatch(piece -> RECEIVED_STATUSES.contains(piece.getReceivingStatus()))) {
+      return PARTIALLY_RECEIVED;
     }
     // Pieces were rolled-back to Expected. In this case we have to check if
     // there is any Received piece in the storage
-    long receivedQty = byPoLine.stream()
+    long receivedQty = piecesByPoLine.stream()
       .filter(piece -> RECEIVED_STATUSES.contains(piece.getReceivingStatus()))
       .count();
-    return Future.succeededFuture(receivedQty == 0 ? AWAITING_RECEIPT : PARTIALLY_RECEIVED);
+    logger.info("calculatePoLineReceiptStatus:: receivedQty: {}", receivedQty);
+    logger.info("calculatePoLineReceiptStatus:: AWAITING_RECEIPT or PARTIALLY_RECEIVED: {}", receivedQty == 0 ? AWAITING_RECEIPT : PARTIALLY_RECEIVED);
```

Comment:
as suggestion earlier, log messages are very hard to understand its purpose. better to add meaningful message and optimize if possible